### ISSUE TITLE
9.1.4.11 - Fokushervorhebung von Formularfeldern

### ISSUE THREAD

**Issue by sweckenmann** [BIK-BITV/BIK-Web-Test#68]
Wenn ich das issue [#1014 der WCAG Working Group](https://github.com/w3c/wcag/issues/1014) richtig lese, ist die blinkende Cursurmarke als Minimal-Anforderung für Formularfelder ausreichend. Das widersprüchliche Beispiel im Understanding-Dokument (Formularfeld mit blauem Rahmen als Fokusindikator, zusäztlich blinkt die Cursormarke) sollte geändert werden (ist vermutlich aufgrund niedriger Priorität noch nicht passiert).

Nun lese ich in 

[G195:](https://www.w3.org/WAI/WCAG21/Techniques/general/G195)

> Example 2: Form Elements
> 
> A Web page includes a form inside a table. The borders of both the table and the form elements are thin, black lines. When focus lands on a form element, the element is outlined with a 5 pixel red line that is partially transparent.
> 
Abgesehen davon, dass wir in der Prüfung nicht auf die Dicke des Fokus-Rahmens eingehen, scheint hier ja dann der Rahmen doch relevant zu sein?

**Comment by detlevhfischer** [comment]
Aus dem good practice Beispiel an G 195 lassen sich (wie aus allen Techniques) aber keine normativen Forderungen ableiten.
Trotzdem ist die Erfüllung nur über Einfügemarke meines Wissens noch nicht durch eine Workgroup Entscheidung abgesichert.

**Comment by sweckenmann** [comment]
Im [WCAG 2.2 Draft / Understanding 1.4.11](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html) ist das kritisierte Beispiel (siehe [WG issue #1014](https://github.com/w3c/wcag/issues/1014) geändert worden:

> While in this case the additional gray (#CCC) outline has an insufficient contrast of 1.6:1 against the white (#FFF) background, the cursor/caret which is displayed when the input receives focus does provide a sufficiently strong visual indication. 

Vielleicht wird G195 ja auch noch angepasst?!

**Comment by johannesFischer84** [comment]
Die Technik G195 scheint sich für mich auf die WCAG 2.2 zu beziehen und nur unglücklich eine Verlinkung zu WCAG 2.1 zu haben.

**Comment by detlevhfischer** [comment]
Das Formular-Beispiel in G195. das Felder mit einem dünnen schwarzen Rand und Fokussierung mit rotem Rahmen beschreibt, ist ja nicht falsch - warum sollte hier ein Beispiel, das 1.4.11 / 2.4.11 nur minimal erfüllt (nur Einfügemarke), genommen werden?
Alle Techniken sind ja nur informelle Resourcen, die im Zweifelsfall als best practice verstanden werden.

**Comment by johannesFischer84** [comment]
Die konkreten Zahlenangaben, ob z. B. die Fokusfläche einer 2-Pixel-Linie um das fokussierte Element herum, entspricht, hat aus meiner Sicht nichts mit 2.4.7 in der aktuellen WCAG 2.1 zu tun, das bezieht sich wohl eher auf die WCAG 2.2. Vom Prinzip her kann die Technik ansonsten natürlich verwendet werden, um bei Erfüllung automatisch 2.4.7 zu erfüllen. Ein gut sichtbarer Fokus ist ja auch das, was ich zu 2.4.7 empfehle, selbst wenn das über die Anforderung von 2.4.7 hinausgeht.

@detlevhfischer Detlev, danke für den Hinweis, dass auch von 1.4.11 auf G195 verwiesen wird. Ich hatte erstmal nur in die Technik geschaut und dort keinen Hinweis auf 1.4.11 gesehen. Vielleicht kommt das noch.

Aber zur Ausgangsfrage: Meiner Meinung nach wäre dann tatsächlich der Fokus-Rahmen trotz Cursor relevant, mindestens mal für 2.4.11, wobei man es wohl auch von 1.4.11 nicht streng trennen könnte. Allerdings ist meine persönliche Meinung auch, dass ein blinkender Cursor als Fokus-Indikator für 1.4.11 nicht ausreichend sein sollte, wenn der Entwickler eine zusätzliche Gestaltung vornimmt. Denn der Cursor ist nur ein kleines Zeichen. Zwar kann der Cursor über die Betriebssystem-Einstellungen beeinflusst werden, aber die Frage besteht, inwiefern das immer zuverlässig klappt. Und die Nutzenden müssen auch zuerst über die Einstellmöglichkeit Bescheid wissen. Wenn ein Entwickler schon die Fokusgestaltung beeinflusst, soll er es auch richtig machen und nicht nur eine halbe Umsetzung mit zu schlechtem Kontrast vornehmen. Aber das ist nur meine persönliche Meinung. Wenn es im W3C anders gesehen wird, nehme ich das natürlich so hin.

Eine Technik ist für mich zwar nicht direkt normativ, aber so ähnlich. Eine Technik ist eine von ggf. mehreren Möglichkeiten, ein Erfolgskriterium zu erfüllen. Gibt es für den jeweiligen Anwendungsfall eine weitere Technik oder eine gleichwertige Art und Weise der Erfüllung ohne Technik, ist das auch in Ordnung. Gibt es so etwas nicht, dann würde ich die Technik schon quasi als verpflichtend ansehen. Sollte sich Technik und Understanding widersprechen, so wie es sich deiner Beschreibung nach, Sonja, für G195 und den Understanding-Draft von 1.4.11 verhält, dann würde ich das als Grauzone sehen. Dann würde ich nur Empfehlungen aussprechen, so wie z. B. auch bei 2.4.4 bei der Verständlichkeit der Linktexte zusammen mit der vorangegangenen Überschrift.
